app/models/nic/base.rb

```diff
@@ -148,6 +151,13 @@ def host_managed?
 
     protected
 
+    def compute_resource_requires_fields
+      if host.compute_resource.present?
```

Comment:
Why is this logic in the NIC and not the host?  Is this just as some hack to skip validation of the MAC address?  I don't think the logic for knowing about compute resources should be here, I'd prefer to lose the skip_mac_validation stuff over misplaced validation logic.